src/model/writer.js

```diff
@@ -1016,6 +1016,35 @@ export default class Writer {
 		}
 	}
 
+	/**
+	 * Disables default gravity behaviour that tries to get attributes from nodes surrounding the caret. When gravity is
+	 * marked as overridden then attributes from the node before the caret won't be taken into consideration while
+	 * updating selection attributes.
+	 *
+	 * For the following model fragment:
+	 *
+	 * 		<$text bold="true" linkHref="url">bar[]</$text><$text bold="true">biz</$text>
+	 *
+	 * Selection attribute keys before override will be equal `[ 'bold', 'linkHref' ]`
+	 * Selection attribute keys after override will be equal `[ 'bold' ]`
+	 *
+	 * As default gravity is automatically restored just after a direct selection change event but this behaviour
```

Comment:
By default the selection's gravity is automatically restored just after a direct selection change ( 
 [[[WHAT IS DIRECT SELECTION CHANGE?]]]  ) but you can pass `customRestore = true` in which case you will have to call ... manually.